src/Sarif.Viewer.VisualStudio/VersionControlParser/GithubVersionControlParser.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text.RegularExpressions;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.Sarif.Viewer.Sarif;
+
+namespace Microsoft.Sarif.Viewer
+{
+    internal class GithubVersionControlParser : IVersionControlParser
+    {
+        private readonly VersionControlDetails details;
+
+        internal GithubVersionControlParser(VersionControlDetails versionControl)
+        {
+            this.details = versionControl;
+        }
+
+        public Uri GetSourceFileUri(string relativeFilePath)
+        {
+            // github link format:
+            // https://github.com/<username>/<reponame>/<blob|tree>/<commitmentid|branch>/path/to/file
+
+            string sourceRelativePath = "blob/";
+            sourceRelativePath += this.details.Branch ?? this.details.RevisionId;
+            sourceRelativePath += "/";
+            sourceRelativePath += relativeFilePath;
+
+            if (Uri.TryCreate(this.details.RepositoryUri, sourceRelativePath, out Uri sourceUri) &&
+                sourceUri.IsHttpScheme())
+            {
+                return new Uri(ConvertToRawPath(sourceUri.ToString()));
+            }
+
+            return null;
+        }
+
+        public string ConvertToRawPath(string url)
```

Comment:
>public string ConvertToRawPath(string url) [](http://example.com/codeflow?start=8&length=42)

make this static

Reply:
I'd like make this method as a method of interface IVersionControlParser.
But C# only supports static interface method in version 8.0 (our version is 7.3). 

Another way is creating a static class member method instead of using interface method.

Reply:
got it